functionframeworks/frameworks.go

```diff
@@ -47,92 +50,102 @@ func registerHTTPFunction(path string, fn func(http.ResponseWriter, *http.Reques
 	return nil
 }
 
-func registerOpenFunction(fn func(*ofctx.OpenFunctionContext, interface{}) int, h *http.ServeMux) error {
+func registerOpenFunction(fn func(*ofctx.OpenFunctionContext, []byte) int, h *http.ServeMux) error {
 	ctx, err := ofctx.GetOpenFunctionContext()
 	if err != nil {
 		return err
 	}
 
-	if *ctx.Input.Enabled {
-		if ctx.Protocol == ofctx.HTTP {
-			h.HandleFunc(ctx.Input.Pattern, func(w http.ResponseWriter, r *http.Request) {
+	if ctx.Protocol == ofctx.HTTP {
```

Comment:
protocol is only used for OpenFuncAsync runtime to specify the protocol between dapr sidecar and function container and it's kind of internal implementation detail and should have default value and shouldn't be exposed to the user actually, right?

So it shouldn't affect non OpenFuncAsync runtime？

Reply:
I define it here as a serving protocol for user functions. So when it is `HTTP` and handled by non-OpenFuncAsync runtime, it means using the HTTP protocol for serving user functions (such as Knative)

Reply:
by default knative is serving via http and there is no protocol defined in function crd https://github.com/OpenFunction/OpenFunction/blob/main/pkg/apis/v1alpha1/function_types.go#L129 .

So protocol is only internal implementation for OpenFuncAsync runtime

Reply:
Ok. According to [Dapr's response](https://github.com/dapr/dapr/issues/2589#issuecomment-749763201), the choice of `grpc` and `http` will depend heavily on the needs of the user function. 

Which protocol should we choose as the default one? `grpc` looks better?

Reply:
yes, grpc is good for dapr and http is the only default option for knative. So it should only apply to OpenFuncAsync runtime

Reply:
and we only need the user to pass the actual input data to function, a function itself needn't know it's passed by grpc or http

Reply:
https://github.com/dapr/dapr/issues/2589#issuecomment-749763201 mentioned it has little difference between grpc or HTTP when using between sidecar and app container. So HTTP is ok too actually

Reply:
The default protocol for OpenFuncAsync runtime has been set to `grpc`